cmd/minikube/cmd/start.go

```diff
@@ -402,29 +401,31 @@ func provisionWithDriver(cmd *cobra.Command, ds registry.DriverState, existing *
 	}, nil
 }
 
-func virtualBoxMacOS13PlusWarning(driverName string) {
-	if !driver.IsVirtualBox(driverName) || !detect.MacOS13Plus() {
+// warnDriverDeprecated prints a small deprecation warning if the selected driver is deprecated
+func warnDriverDeprecated(driverName string) {
+	if driverName == "" {
 		return
 	}
-	out.WarningT(`Due to changes in macOS 13+ minikube doesn't currently support VirtualBox. You can use alternative drivers such as 'vfkit', 'qemu', or 'docker'.
-    https://minikube.sigs.k8s.io/docs/drivers/vfkit/
-    https://minikube.sigs.k8s.io/docs/drivers/qemu/
-    https://minikube.sigs.k8s.io/docs/drivers/docker/
-    For more details on the issue see: https://github.com/kubernetes/minikube/issues/15274
-`)
-}
 
-// hyperkitDeprecationWarning prints a deprecation warning for the hyperkit driver
-func hyperkitDeprecationWarning(driverName string) {
-	if !driver.IsHyperKit(driverName) {
-		return
+	for _, d := range registry.List() {
+		if d.Priority != registry.Deprecated {
+			continue
+		}
+
+		// match driver name
+		if d.Name == driverName {
+			out.WarningT("The '{{.name}}' driver is deprecated and will be removed in a future release.\n    You can use alternative drivers: {{.drivers}}.", out.V{"name": d.Name, "drivers": strings.Join(driver.SupportedDrivers(), ", ")})
```

Comment:
SupportedDrivers() return results based on current OS? Is it sorted by priority?

Previously we used: 

   You can use alternative drivers such as 'vfkit', 'qemu', or 'docker'

This is the wanted order, using driver priority (vfkit is the preferred driver) and preferring vm drivers since on macOS all drivers need a vm. We want to keep this order. If SupportedDrivers() does not return the drivers in the right order for the current OS, we need to improve it before using it here.

Reply:
The order is wrong - vfkit is the preferred driver and should be first, not qemu. This is the native driver for macOS. qemu on macOS is poorly maintained and should not be the preferred driver.

Also the hyperkit driver appears in the list of alternatives which is wrong - the hyperkit driver is not an alternative to itself.

We need to create AlternativeDrivers() function - SupportedDrivers() is not the right way.